### PR TITLE
optimization of issue summary state

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of this PR

Add one of the following kinds:
/kind bugfix

#### What this PR does / why we need it:
optimization of  query issue summary state
reduce the query of the sql statement

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=257960&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=BUG)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Fix the bug that cost long time when query iteration            |
| 🇨🇳 中文    |   修复了获取迭代耗时太久           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: apistructs/iteration.go [hunk 1]
```diff
@@ -147,6 +147,10 @@ type ISummary struct {
...
 
 type ISummaryState struct {
-	Done   int `json:"done"`
-	UnDone int `json:"undone"`
+	Done        int       `json:"done"`
+	UnDone      int       `json:"undone"`
+	Total       int       `json:"total,omitempty"`
+	IssueType   IssueType `json:"issue_type,omitempty"`
+	State       int64     `json:"state,omitempty"`
+	IterationID int64     `json:"iteration_id,omitempty"`
```
Comment (Effet) [inline apistructs/iteration.go:155]: does necessary to export these fields to api?

**Review by Effet — CHANGES_REQUESTED** [review]

File: apistructs/iteration.go (hunk 1 above)
Comment (Effet) [reply apistructs/iteration.go:155]: you need to define a new model for dao.query only, not share the struct with api

**Review by Effet — CHANGES_REQUESTED** [review]

File: apistructs/iteration.go (hunk 1 above)
Comment (chengjoey) [reply apistructs/iteration.go]: adjusted

File: modules/dop/services/iteration/iteration.go [hunk 2]
```diff
@@ -210,3 +210,93 @@ func (itr *Iteration) GetIssueSummary(iterationID int64, projectID uint64) (apis
...
+	}
+	return nil
+}
+
+func (itr *Iteration) getDoneStateIDSByType(projectID uint64, issueType apistructs.IssueType) ([]int64, error) {
+	states, err := itr.db.GetIssuesStatesByProjectID(projectID, issueType)
+	if err != nil {
+		return nil, err
+	}
+	stateIDS := make([]int64, 0)
```
Comment (Effet) [inline modules/dop/services/iteration/iteration.go:277]: -> stateIDs

**Review by Effet — APPROVED** [review]

**Pull request merged**